src/journal/journal-file.c

```diff
@@ -123,26 +132,48 @@ static void journal_file_set_offline_internal(JournalFile *f) {
                                 continue;
                         break;
 
-                case OFFLINE_SYNCING:
-                        (void) fsync(f->fd);
+                case OFFLINE_SYNCING: {
+                        int r;
+
+                        /* OFFLINE_AGAIN_* can bring us back here, prevent another fsync() if a previous one failed */
+                        if (status < 0) {
```

Comment:
Handling in both conditionals is the same, they should be combined. What about:
```C
if (status >= 0)
    if (fsync(f->fd) < 0)
         status = -errno;
if (status < 0) {
     ...
}
```

Reply:
That's a reasonable deduplication of the code, I think I was just in a "keep things plain and obvious here" mode.